clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/front50/Front50ArtifactCredentials.java

```diff
@@ -0,0 +1,82 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.artifacts.front50;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spinnaker.clouddriver.artifacts.config.ArtifactCredentials;
+import com.netflix.spinnaker.clouddriver.core.services.Front50Service;
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang.StringUtils;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+@Slf4j
+@Data
+public class Front50ArtifactCredentials implements ArtifactCredentials {
+  private final String name = "front50ArtifactCredentials";
+  // TODO(jacobkiefer): Delegate to specialized artifact handlers once we add more types.
+  private final List<String> types = Arrays.asList("front50/pipelineTemplate");
+
+  private Front50Service front50Service;
+  private ObjectMapper objectMapper;
+
+  Front50ArtifactCredentials(ObjectMapper objectMapper,
+                             Front50Service front50Service) {
+    this.objectMapper = objectMapper;
+    this.front50Service = front50Service;
+  }
+
+  @Override
+  public InputStream download(Artifact artifact) throws IOException {
+    if (StringUtils.isEmpty(artifact.getName())) {
+      throw new IllegalArgumentException("'front50/pipelineTemplate' artifacts must be specified with a 'name', got: " + artifact);
+    }
+
+    String pipelineTemplateId = artifact.getName();
+    String version = artifact.getVersion();
+    Map pipelineTemplate;
+    if (StringUtils.isEmpty(version)) {
+      pipelineTemplate = front50Service.getV2PipelineTemplate(pipelineTemplateId, "", "");
+    } else if (version.startsWith("sha256:")) { // Version is specifying a digest.
+      pipelineTemplate = front50Service.getV2PipelineTemplate(pipelineTemplateId, "", version);
+    } else {
+      pipelineTemplate = front50Service.getV2PipelineTemplate(pipelineTemplateId, version, "");
+    }
+
+    return new ByteArrayInputStream(objectMapper.writeValueAsBytes(pipelineTemplate));
+  }
+
+  @JsonIgnore
+  @Override
+  public List<String> getArtifactNames() {
+    return front50Service.listV2PipelineTemplateIds();
+  }
+
+  @Override
+  @JsonIgnore
+  public List<String> getArtifactVersions(String artifactName) {
```

Comment:
It might be better not to override this, and to leave it to fall back to the "not implemented" exception in the parent for clearer errors if this ever gets used

Reply:
Done.